### ISSUE TITLE
resolveNewValue picker error

### ISSUE THREAD

**Issue by Ridlsan** [OfficeDev/office-ui-fabric-react#3457]
There is code in components/pickers/BasePicker.tsx in resolveNewValue :
```
  this.setState({
      suggestionsLoading: false,
      suggestedDisplayValue: itemValue,
      suggestionsVisible: this.input.value !== '' && this.input.inputElement === document.activeElement
});
```

If u limit picker with itemLimit, when u select last item input is removed, so this code throughs error (TypeError: this.input is null). Easy fix is 
```
  this.setState({
      suggestionsLoading: false,
      suggestedDisplayValue: itemValue,
      suggestionsVisible: this.input && this.input.value !== '' && this.input.inputElement === document.activeElement
});
```

**Comment by micahgodbolt** [comment]
Is this a fix you'd be willing to submit? I'd be happy to review your submission.

**Comment by ohritz** [comment]
I think, maybe just maybe this will fix issues #3483, #3520, #3297 aswell.

Full disclosure, I implemented the itemLimit property, with conditional rendering of the input on itemLimit being reached, in effect causing this bug.
There are more places in the basePicker that assume the input is present, but not sure how many are in the execution path when its not present.

**Comment by Ridlsan** [comment]
Yeah, seems like this patch would fix it
